streams/test-utils/src/main/java/org/apache/kafka/streams/processor/MockProcessorContext.java

```diff
@@ -324,6 +330,14 @@ public long offset() {
         return offset;
     }
 
+    @Override
+    public Headers headers() {
+        if (headers == null) {
+            throw new IllegalStateException("Headers must be set before use via setRecordMetadata() or setHeaders()");
```

Comment:
Why is this required, ie, why is returning `null` bad?

Reply:
You're right. Fixing it.